### DISCUSSION TITLE
rabbitmq 3.7.5 Erlang 20.3.7  memory leak reader process.

### DISCUSSION THREAD

**Discussion by SSANSH** [rabbitmq/rabbitmq-server#4224] [Other]
Hi, 
I am using rabbitmq 3.7.5 Erlang 20.3.7 (3 nodes),
Cluster has run correctly work for a long time except for last few days.
Since last few days, I experiment an issue about memory consumption on only two node. 
Memory never stop to grown like a memory leak.
 
We enable the plugin top on rabbitmq-server and we saw some reader process which looks like stuck in erlang memory.
I have this feeling because if I take connection on description column, I can see the connection on rabbitmq side (ui), however consumer tag indicates to me an old client on my platform which don’t exist anymore.
I check on client side and I have nothing about this connection (tcp), I check also on server side and this is the same (tcp).
I saw that connection has no channel setup and no consumer, that’s make sense because client not exist in client side.
When I try to close the connection by ui or by rabbitmqctl command, I cannot, ui et command freeze.
 
Do you know when this case can arrive, this looks like a bug of rabbitmq?
Connection looks like to be in ghost mode.
I know that the best solution is to upgrade my cluster, but I would like to know if a defect exist on rabbitmq side and if it was fix.
At this point for fixing the memory consumption we restart pysically each node of my cluster, and the memory decreased drastically.

**Comment by michaelklishin** [comment]
Both versions are out of date and [no longer supported](https://www.rabbitmq.com/versions.html).
